privacyidea.py

```diff
@@ -2,131 +2,241 @@
 from org.gluu.oxauth.service import AuthenticationService, SessionIdService
 from org.gluu.model.custom.script.type.auth import PersonAuthenticationType
 from org.gluu.service.cdi.util import CdiUtil
+from org.gluu.oxauth.service import UserService
+from org.gluu.util import StringHelper
 
+from javax.faces.application import FacesMessage
+from org.gluu.jsf2.message import FacesMessages
 
+from java.util import Arrays
 
-# TODO: Import the privacyIDEA Java SDK
 
-import java
+def logFromSDK(message):
+    print("privacyIDEA. JavaSDK: " + message)
 
 
 class PersonAuthentication(PersonAuthenticationType):
+
     def __init__(self, currentTimeMillis):
+        print("__init__")
         self.currentTimeMillis = currentTimeMillis
-        self.identity = CdiUtil.bean(Identity)
-        self.sessionIdService = CdiUtil.bean(SessionIdService)
-        self.authenticationService = CdiUtil.bean(AuthenticationService)
 
     def init(self, configurationAttributes):
-        print "privacyIDEA. Initialization"
+        print("privacyIDEA. init")
+
+        sdk_path = "/opt/java_sdk.jar"
+        if configurationAttributes.contains("sdk_path"):
+            sdk_path = configurationAttributes.get("sdk_path").getValue2()
+
+        import sys
```

Comment:
Please put the import at the beginning of the file unless it's necessary to have it here.

Reply:
ok